homework/hw9/task3_tokotoko.py

```diff
@@ -27,10 +27,13 @@ def universal_file_counter(
     count = 0
     for file in dir_path.glob("*." + file_extension):
         with open(file) as f:
-            count += (
-                sum(len(tokenizer(line)) for line in f)
-                if tokenizer
-                else sum(1 for _ in f)
-            )
-
+            count += sum(len(tokenizer(line)) if tokenizer else 1 for line in f)
     return count
+
+
+"""token = f(x) if tokenizier else 1
+    count += sum(token(line) for line in f)
+
+def f(x):
+    return len(tokenizer(x))
+    """
```

Comment:
to be honest, not get the idea what the best way to implement it, instead of one liner we can do token, and then sum with token, i think that's what you meant.

maybe you can also write it as 
token = lambda x: len(tokenizer(x)) if tokenizer else 1
count += sum(token(line) for line in f)

Reply:
Current solution seems much simpler. Also, here we can read data from file without any logic and then use lines for our purposes:

```
    count = 0
    for file in dir_path.glob("*." + file_extension):
        with open(file) as f:
            lines = f.readlines()
        count += sum(len(tokenizer(line)) if tokenizer else 1 for line in lines)
    return count
```